main.rb

```diff
@@ -0,0 +1,41 @@
+def bubble_sort(arr)
+  i = 0
+  k = 0
+  flag = false
+
+  while i < (arr.length - 1)
+    while k < (arr.length - 1 - i)
+      if arr[k] > arr[k + 1]
+        arr[k], arr[k + 1] = arr[k + 1], arr[k]
+        flag = true
+      end
+      k += 1
+    end
+    break if flag == false
+
+    i += 1
+  end
+  arr
+end
```

Comment:
I tried to execute the method with an unsorted array. But did not receive the expected result.
```ruby
p bubble_sort([9,2,1,6,5,4,8,10,7,3]) # The result I got [2, 1, 6, 5, 4, 8, 9, 7, 3, 10]
```